### PR TITLE
update algebraic expression operands dim as a result of a prev write

### PR DESCRIPTION
**Pull request by swilly22**
#836 
merge followed by a second merge position us in a write read write situation,
in this case early creation of both iterator and algebraic expressions might be out of sync with data written in the first merge.

### CODE REVIEW

File: src/execution_plan/ops/shared/create_functions.c [hunk 1]
```diff
@@ -136,17 +136,23 @@ PendingCreations NewPendingCreationsContainer(ResultSetStatistics *stats, NodeCr
 // Lock the graph and commit all changes introduced by the operation.
 void CommitNewEntities(OpBase *op, PendingCreations *pending) {
 	Graph *g = QueryCtx_GetGraph();
+	GraphContext *gc = QueryCtx_GetGraphCtx();
```
Comment (jeffreylovitz) [inline src/execution_plan/ops/shared/create_functions.c]: Unused variable.

File: src/arithmetic/algebraic_expression.c [hunk 2]
```diff
@@ -674,6 +676,53 @@ void AlgebraicExpression_Free(AlgebraicExpression *ae) {
...
+			else operand->operand = Graph_GetLabelMatrix(g, s->id);
+		} else {
+			// Operand represents an edge.
+			if(!label) {
+				operand->operand = Graph_GetAdjacencyMatrix(g);
+				continue;
+			}
+
+			s = GraphContext_GetSchema(gc, label, SCHEMA_EDGE);
+			if(!s) {
```
Comment (jeffreylovitz) [inline src/arithmetic/algebraic_expression.c]: Nit, but could be a 2-liner.

File: src/arithmetic/algebraic_expression.c [hunk 3]
```diff
@@ -674,6 +676,53 @@ void AlgebraicExpression_Free(AlgebraicExpression *ae) {
...
+		if(operand->free) {
+			GrB_Index dim = Graph_RequiredMatrixDim(g);
+			// Ok to simply resize.
+			GxB_Matrix_resize(operand->operand, dim, dim);
+			continue;
+		}
+
+		if(operand->diagonal) {
+			// Operand represents a node.
+			s = GraphContext_GetSchema(gc, label, SCHEMA_NODE);
```
Comment (jeffreylovitz) [inline src/arithmetic/algebraic_expression.c]: `GraphContext_GetSchema` (called here and at 718) rely on strcmp loops, but given that I'd expect a very small N for schema count, that doesn't necessarily need to be optimized. Still, we could cache the ID on the `AlgebraicExpressionOperand` with or as a replacement to `operand->label`, and only make this call while the schema doesn't yet exist.

File: src/execution_plan/ops/shared/create_functions.c (hunk 1 above)
Comment (swilly22) [reply src/execution_plan/ops/shared/create_functions.c]: 👍

File: src/arithmetic/algebraic_expression.c (hunk 2 above)
Comment (swilly22) [reply src/arithmetic/algebraic_expression.c]: 👍

File: src/arithmetic/algebraic_expression.c (hunk 3 above)
Comment (swilly22) [reply src/arithmetic/algebraic_expression.c]: Yeah that would be an optimisation. we can also switch to rax.

**Pull request merged**